front/pages/no-workspace.tsx

```diff
@@ -0,0 +1,144 @@
+import {
+  BarHeader,
+  Button,
+  Icon,
+  LogoSquareColorLogo,
+  Page,
+} from "@dust-tt/sparkle";
+import type { InferGetServerSidePropsType } from "next";
+import { useRouter } from "next/router";
+
+import { getSession, getUserFromSession } from "@app/lib/auth";
+import { Membership, Workspace } from "@app/lib/models";
+import logger from "@app/logger/logger";
+import { withGetServerSidePropsLogging } from "@app/logger/withlogging";
+
+export const getServerSideProps = withGetServerSidePropsLogging<{
+  revokedWorkspaceName?: string;
+  userFirstName: string;
+}>(async (context) => {
+  const session = await getSession(context.req, context.res);
+  const user = await getUserFromSession(session);
+
+  if (!user) {
+    return {
+      notFound: true,
+    };
+  }
+
+  const memberships = await Membership.findAll({
+    where: { userId: user.id },
+  });
+
+  if (!memberships.length) {
+    const message = "Unreachable: user has no memberships";
+    logger.error({ userId: user.id, panic: true }, message);
+    throw new Error(message);
+  }
+
+  if (user.workspaces.length) {
+    const message = "Unreachable: user already has a workspace";
+    logger.error({ userId: user.id, panic: true }, message);
+    return {
+      notFound: true,
+    };
+  }
+
+  const revokedWorkspaceId = memberships[0].workspaceId;
+
+  const workspace = await Workspace.findByPk(revokedWorkspaceId);
+
+  if (!workspace) {
+    const message = "Unreachable: workspace not found";
+    logger.error(
+      { userId: user.id, workspaceId: revokedWorkspaceId, panic: true },
+      message
+    );
+    throw new Error(message);
+  }
+
+  return {
+    props: {
+      revokedWorkspaceName: workspace.name,
+      userFirstName: user.firstName,
+    },
+  };
+});
+
+export default function NoWorkspace({
+  revokedWorkspaceName,
+  userFirstName,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const router = useRouter();
+
+  const onCreateWorkspace = async () => {
+    const res = await fetch("/api/create-new-workspace", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    if (!res.ok) {
+      console.error("Failed to create new workspace");
```

Comment:
Should we keep this log?

Reply:
Can replace with a sendNotification maybe